pkg/account/component_test.go

```diff
@@ -67,37 +67,79 @@ func TestUpdatePassword(t *testing.T) {
 	ctx = context.WithValue(ctx, cs.CtContextUserID, userID)
 	ctx = context.WithValue(ctx, cs.CtContextUsername, username)
 
+	oldPasswd := "prev10u5"
+	newPasswd := "a p@55w0rd"
+	anError := errors.New("any error")
+	temporary := false
+	setPassword := kc.CredentialRepresentation{
+		Type:      cs.ToStringPtr("password"),
+		Value:     cs.ToStringPtr(newPasswd),
+		Temporary: &temporary,
+	}
+
 	t.Run("Update password: no change", func(t *testing.T) {
-		oldPasswd := "a p@55w0rd"
-		err := component.UpdatePassword(ctx, oldPasswd, oldPasswd, oldPasswd)
+		oldPassword := "a p@55w0rd"
+		err := component.UpdatePassword(ctx, oldPassword, oldPassword, oldPassword)
 
 		assert.NotNil(t, err)
 	})
 
 	t.Run("Update password: bad confirm", func(t *testing.T) {
-		oldPasswd := "prev10u5"
-		newPasswd := "a p@55w0rd"
 		confirmPasswd := "bad one"
 		err := component.UpdatePassword(ctx, oldPasswd, newPasswd, confirmPasswd)
 
 		assert.NotNil(t, err)
 	})
 
 	t.Run("Update password: success", func(t *testing.T) {
-		anyError := errors.New("any error")
-		oldPasswd := "prev10u5"
-		newPasswd := "a p@55w0rd"
 		confirmPasswd := "a p@55w0rd"
 		mocks.keycloakAccountClient.EXPECT().UpdatePassword(accessToken, realm, oldPasswd, newPasswd, confirmPasswd).Return("", nil)
 		mocks.auditEventsReporterModule.EXPECT().ReportEvent(gomock.Any(), gomock.Any())
 		mocks.keycloakAccountClient.EXPECT().SendEmail(accessToken, realm, emailTemplateUpdatedPassword, emailSubjectUpdatedPassword, nil, gomock.Any()).Return(nil)
 		mocks.auditEventsReporterModule.EXPECT().ReportEvent(gomock.Any(), gomock.Any())
-		mocks.keycloakTechnicalClient.EXPECT().LogoutAllSessions(gomock.Any(), realm, userID).Return(anyError)
+		mocks.keycloakTechnicalClient.EXPECT().LogoutAllSessions(gomock.Any(), realm, userID).Return(anError)
 
 		err := component.UpdatePassword(ctx, oldPasswd, newPasswd, confirmPasswd)
 
 		assert.Nil(t, err)
 	})
+
+	t.Run("Set password: GetCredentials fails", func(t *testing.T) {
+		mocks.keycloakAccountClient.EXPECT().GetCredentials(accessToken, realm).Return([]kc.CredentialRepresentation{}, anError)
+
+		err := component.UpdatePassword(ctx, "", newPasswd, newPasswd)
+		assert.NotNil(t, err)
+	})
+
+	t.Run("Set password: user has a current password", func(t *testing.T) {
+		cred := kc.CredentialRepresentation{
+			Type: cs.ToStringPtr("password"),
```

Comment:
ToStringPtr is a complex processing used for incoming data according to a given user profile. We don't really need this so complex processing... but why not.

Reply:
You're right, I changed it